src/pages/Settings.vue

```diff
@@ -6,6 +6,26 @@
             </h1>
 
             <div class="shadow-box">
+                <div class="row">
+                    <div class="settings-menu">
+                        <div v-for="(item, key) in subMenus"
+                             :key="key"
+                             class="menu-item"
+                             :class="{ active: currentSubMenu == key }"
+                             @click="currentSubMenu = key"
+                        >
```

Comment:
Using vue-router would be better, since vue-router handled browser history and the back button action gracefully. Also user could go to the page with a direct link like `http://localhost/settings/appearance` 

`<router-link>`